src/lib/circuit_value.ts

```diff
@@ -569,6 +568,22 @@ function provable<A>(
     );
     return Object.fromEntries(keys.map((k, i) => [k, values[i]]));
   }
+  function fromJSON(typeObj: any, json: any, isToplevel = false): any {
```

Comment:
a major theme of this PR is that `fromJSON` functions were added to all provables